app-arch/tar/patches/tar-1.29.patchset

```diff
@@ -5,26 +5,23 @@ Subject: gcc2 patch
 
 
 diff --git a/lib/wordsplit.h b/lib/wordsplit.h
-index d64cf2f..c93f53d 100644
+index e0d2559..acf1f56 100644
 --- a/lib/wordsplit.h
 +++ b/lib/wordsplit.h
-@@ -34,9 +34,9 @@ struct wordsplit
-   const char *ws_escape;
-   void (*ws_alloc_die) (struct wordsplit * wsp);
-   void (*ws_error) (const char *, ...)
--                   __attribute__ ((__format__ (__printf__, 1, 2)));
-+;
-   void (*ws_debug) (const char *, ...)
--                   __attribute__ ((__format__ (__printf__, 1, 2)));
-+;
+@@ -22,7 +22,7 @@
  
-   const char **ws_env;
-   const char *(*ws_getvar) (const char *, size_t, void *);
+ #include <stddef.h>
+ 
+-#if 2 < __GNUC__ + (7 <= __GNUC_MINOR__)
++#if 2 < __GNUC__ + (7 <= __GNUC_MINOR__) - __HAIKU__
```

Comment:
I've just retried building tar 1.29 on x86_gcc2 without the first chunk of the patch, i.e. the chunk with the tiny hack. The result is:
```
  CC       wordsplit.o
In file included from /sources/tar-1.29/lib/wordsplit.c:40:
/sources/tar-1.29/lib/wordsplit.h:43: argument format specified for non-function `ws_error'
/sources/tar-1.29/lib/wordsplit.h:45: argument format specified for non-function `ws_debug'
Makefile:1287: recipe for target 'wordsplit.o' failed
```
So I reworked that chunk to make it work, but avoiding the use of **`__HAIKU__`**.
I don't know if gcc > 2.95 is the correct criteria, but it is sufficient to discriminate between the version of gcc2 we have in x86_gcc2 and the one we have in x86 and x86_64.